### DISCUSSION TITLE
how to remove plugins object types from the permissions page?

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#17261] [Q&A] [answered]
I installed phonebox_plugin : https://github.com/iDebugAll/phonebox_plugin

Wasn't good fit for our use case, so I uninstalled it.
I followed the procedures here: https://netboxlabs.com/docs/netbox/en/stable/plugins/removal/

including dropping the tables and removing the django migration records.
Restarted netbox, but when I go to the 'permissions' section I notice that the following are still listed in object types.

![image](https://github.com/user-attachments/assets/7fe2909b-1c51-4b46-b6ff-45c577a32b0d)

How do I get rid of these from the permissions editor?

**Comment by candlerb** [comment]
My first guess would be this is showing all entries from the django_content_type table.

```
# /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell
...
>>> from django.contrib.contenttypes.models import ContentType
>>> ContentType.objects.all()
<QuerySet [<ContentType: logentry>, <ContentType: Authentication and Authorization | group>,
...
>>> ContentType.objects.filter(model="interface")
<QuerySet [<ContentType: DCIM | interface>]>
>>>
```

**Reply by ghost** [reply]
im sorry, im unable to follow your post. is what you posted instructions on how to remove things? or just providing further context?

**Reply by candlerb** [reply]
I'm showing you how to isolate the problem. I have a theory, which is that the models still exist in the `django_content_type` table; you can either confirm or deny that.

If it's confirmed, then you can try deleting the corresponding objects. In nbshell you'd use the [`delete()`](https://stackoverflow.com/questions/3805958/how-to-delete-a-record-in-django-models) method on an object instance or a queryset. In psql you'd use `DELETE FROM django_content_type WHERE <suitable condition>`. Pick whichever you're most comfortable with.

Make sure you have a full backup of your database before starting this, as documented [here](https://netboxlabs.com/docs/netbox/en/stable/administration/replicating-netbox/).

**Comment by ghost** [comment]
I'm not sure how running the commands in your first reply confirms anything about the issue above. I've never used nbshell before.

`user@dc1-netbox-p01:~$ sudo /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell`
`### NetBox interactive shell (dc1-netbox-p01)`
`### Python 3.10.12 | Django 5.0.7 | NetBox 4.0.8`
`### lsmodels() will show available models. Use help(<model>) for more info.`
`>>> from django.contrib.contenttypes.models import ContentType`
`>>> ContentType.objects.all()`
`<QuerySet [<ContentType: Extras | webhook>, <ContentType: Extras | script>, <ContentType: Extras | script module>, <ContentType: Core | managed file>, <ContentType: Authentication and Authorization | permission>, <ContentType: Authentication and Authorization | group>, <ContentType: Content Types | content type>, <ContentType: Sessions | session>, <ContentType: Python Social Auth | association>, <ContentType: Python Social Auth | code>, <ContentType: Python Social Auth | nonce>, <ContentType: Python Social Auth | user social auth>, <ContentType: Python Social Auth | partial>, <ContentType: Taggit | tag>, <ContentType: Taggit | tagged item>, <ContentType: Core | data source>, <ContentType: Core | data file>, <ContentType: Core | auto sync record>, <ContentType: Core | job>, <ContentType: Core | object type>, '...(remaining elements truncated)...']>`
`>>> ContentType.objects.filter(model="interface")`
`<QuerySet [<ContentType: DCIM | interface>]>`
`>>> ContentType.objects.filter(model="number")`
`<QuerySet [<ContentType: number>]>`
`>>> ContentType.objects.filter(model="voicecircuit")`
`<QuerySet [<ContentType: voicecircuit>]>`
`>>> ContentType.objects.filter(model="phonebox_plugin")`
`<QuerySet []>`
`>>>`

**Comment by jsenecal** [comment, marked as answer]
Try that:
```shell
sudo /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell
>>> from django.contrib.contenttypes.models import ContentType
>>> ct_qs = ContentType.objects.filter(app_label__icontains="phone")
>>> for ct in ct_qs:
...          print(ct)
### ^^^ These will be removed, make sure its ok
>>> ct_qs.delete()
```

**Comment by ghost** [comment]
`>>> from django.contrib.contenttypes.models import ContentType`
`>>> ct_qs = ContentType.objects.filter(app_label__icontains="phone")`
`>>> for ct in ct_qs: print(ct)`
`...`
`number`
`voicecircuit`
`>>> ct_qs.delete()`
`(12, {'auth.Permission': 8, 'users.ObjectPermission_object_types': 2, 'contenttypes.ContentType': 2})`
`>>>`
